### ISSUE TITLE
[GL backend] shader translation fails

### ISSUE THREAD

**Issue by dhardy** [gfx-rs/naga#1085]
Trying to run KAS via the GL backend succeeds after disabling three of the graphics pipelines (flat_round, shaded_round, shaded_square). When re-enabling `flat_round` this error occurs:
```
[2021-07-13T16:09:40Z INFO  gfx_backend_gl::device]     Compiled shader 2
[2021-07-13T16:09:40Z INFO  gfx_backend_gl::device]     Compiled shader 3
[2021-07-13T16:09:40Z INFO  gfx_backend_gl::device]     Linked program 1
[2021-07-13T16:09:40Z ERROR gfx_backend_gl] [ShaderCompiler/Error] ID 2 : 0:33(13): error: initializer of type int cannot be assigned to variable of type float
[2021-07-13T16:09:40Z ERROR gfx_backend_gl] [ShaderCompiler/Error] ID 3 : 0:35(14): error: initializer of type int cannot be assigned to variable of type float
[2021-07-13T16:09:40Z ERROR gfx_backend_gl] [ShaderCompiler/Other] ID 4 : 0:62(18): warning: `a1_1' used uninitialized
[2021-07-13T16:09:40Z INFO  gfx_backend_gl::device]     Compiled shader 5
[2021-07-13T16:09:40Z WARN  wgpu::backend::direct] Shader translation error for stage VERTEX: Shader module failed to compile: 0:33(13): error: initializer of type int cannot be assigned to variable of type float
    0:35(14): error: initializer of type int cannot be assigned to variable of type float
    0:62(18): warning: `a1_1' used uninitialized
     shader compilation failed: Vertex
[2021-07-13T16:09:40Z WARN  wgpu::backend::direct] Please report it to https://github.com/gfx-rs/naga
[2021-07-13T16:09:40Z WARN  wgpu::backend::direct] Try enabling `wgpu/cross` feature as a workaround.
[2021-07-13T16:09:40Z ERROR wgpu::backend::direct] wgpu error: Validation Error
```
I believe [this shader](https://github.com/kas-gui/kas/blob/c5212b2c151ab935438e85bb9c4da1f24aeb1286/kas-wgpu/src/draw/shaders/scaled_122.vert) is the one mentioned (though it would be clearer if there was a label).

`a1` is `layout(..) in float a1`; the label `a1_1` used in the warning implies the translator thinks this is a vec?

**Comment by dhardy** [comment]
I believe the issue is lack of support for `float` inputs to the shader. Switching to `vec2` input "solves" this, but...

Given
```
layout(location = 1) in vec2 inner;
```
in the fragment shader, if `inner.y` is never used, I get the same error:
```
        Internal error in FRAGMENT shader: Shader module failed to compile: 0:5(16): error: initializer of type int cannot be assigned to variable of type float
    0:30(18): warning: `inner1' used uninitialized
     shader compilation failed: Fragment
```
Adding a read (e.g. `float dummy = inner.y;`) works around this error.

Using the `wgpu/cross` feature works around this (as well as some other bug causing some UI elements to render black).

**Comment by kvark** [comment]
Could you run with "debug" log level to see the generated GLSL?

**Comment by dhardy** [comment]
[gallery.log.gz](https://github.com/gfx-rs/naga/files/6854489/gallery.log.gz)

**Comment by dhardy** [comment]
BTW that was after #1119.

**Comment by dhardy** [comment]
Yes, looks like this is fixed by #1119 to me.
